src/osd/OSDMap.cc

```diff
@@ -4605,21 +4644,71 @@ void OSDMap::check_health(health_check_map_t *checks) const
 
   // POOL_FULL
   {
-    list<string> detail;
+    enum {
+      WARN_POOL_FULL,
+      WARN_POOL_SIZE,
+      WARN_POOL_MIN_SIZE,
+      WARN_POOL_MAX_TYPE,
```

Comment:
Is this enum helpful?  We're only ever referencing vector positions explicitly; we don't programatically loop over them.  May as well just do
<pre>
list<string> detail_full, detail_size, detail_min_size, detail_max_type;
</pre>
?